internal/monitors/manager.go

```diff
@@ -314,26 +314,39 @@ func (mm *MonitorManager) findConfigForMonitorAndRun(endpoint services.Endpoint)
 // endpoint may be nil for static monitors
 func (mm *MonitorManager) createAndConfigureNewMonitor(config config.MonitorCustomConfig, endpoint services.Endpoint) error {
 	id := types.MonitorID(mm.idGenerator())
+	coreConfig := config.MonitorConfigCore()
+	monitorType := coreConfig.Type
 
 	log.WithFields(log.Fields{
-		"monitorType":   config.MonitorConfigCore().Type,
-		"discoveryRule": config.MonitorConfigCore().DiscoveryRule,
+		"monitorType":   monitorType,
+		"discoveryRule": coreConfig.DiscoveryRule,
 		"monitorID":     id,
 	}).Info("Creating new monitor")
 
 	instance := newMonitor(config.MonitorConfigCore().Type, id)
 	if instance == nil {
-		return errors.Errorf("Could not create new monitor of type %s", config.MonitorConfigCore().Type)
+		return errors.Errorf("Could not create new monitor of type %s", monitorType)
+	}
+
+	metadata, ok := MonitorMetadatas[monitorType]
+	if !ok {
+		return errors.Errorf("could not find monitor metadata of type %s", monitorType)
```

Comment:
If this is missing, doesn't it indicate something is majorly wrong with the logic that registers monitors?  If so, it should just panic.